### PR TITLE
Accept DAI as payment WIP

### PR DESCRIPTION
**Pull request by woeltjen**
## Description
Integrate Uniswap into our payment flow to allow DAI as a payment option.

## How to Test
- [ ] Visit XXX
- [ ] Click Y

Which devices did you test on?
- [ ] Chrome on Mac
- [ ] Chrome on PC
- [ ] Firefox on Mac
- [ ] Firefox on PC
- [ ] Safari iPhone
- [ ] Chrome Android

## REVIEWERS:
Check against these principles:

### High level
Does this code need to be written?
What are the alternatives?
Will this implementation become a support issue?
How much error margin does this solution have?

### Code
* Does the code follow industry standards?
JS: https://github.com/airbnb/javascript
React: https://github.com/airbnb/javascript/tree/master/react
https://github.com/vasanthk/react-bits
Documentation headers: http://usejsdoc.org/index.html

* Is there duplicated code? Can it be refactored into a shared method?
* Is the code consistent with our project?
* Are there unit tests? Do they test the states?
* Is the person refactoring another developer's code? If possible, did the original developer approve?

### Variables/Naming:
* Would the variable type led to future edge cases?
* Are the variable naming clear? Would the value contain something other than what the name describes.

### Security
* Can this be hacked or abused by the user?

## Further Work
Expect three significant follow-up tasks, in roughly this order:

1. Smooth out UX so the swap step isn't seen by the user
2. Generalize to support for pay-with-any-ERC-20
3. Generalize more to support pay-*to*-any-ERC-20

## Learnings
Storing our ABIs in beautified TypeScript files makes them less portable; it adds some `export` boilerplate and uses single-quotes, but everywhere else that will use an ABI expects JSON

### CODE REVIEW

File: src/utils/uniswap.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import Web3 from 'web3'; // We're only importing the types
+import UNISWAP_FACTORY_ABI from 'ABIs/uniswapFactory.json';
+import UNISWAP_EXCHANGE_ABI from 'ABIs/uniswapExchange.json';
+
+export default class Uniswap {
```
[the remark this answers is not in the document]
Comment (tc) [reply src/utils/uniswap.ts]: I'm fine with skipping ahead if there's nothing to be learned from ``1-3`` especially if all the work is being done at the smart contract level which can be tested with direct contract calls.

**Comment by woeltjen** [comment]
Closing in favor of #60
